playbooks/generic-tempest-output-to-junit.yml

```diff
@@ -0,0 +1,83 @@
+---
+- hosts: soc-deployer
+  gather_facts: yes
+  tasks:
+    - name: Run container to attach tempest pvc
+      shell: |
+        set pipefail -o
+        cat <<EOF | kubectl -n openstack apply -f -
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: tempest-data
+        spec:
+          containers:
+          - name: tempest-data
+            image: opensuse/leap
+            command: ["/bin/bash", "-ec", "while :; do echo '.'; sleep 5 ; done"]
+            volumeMounts:
+            - name: mypvc
+              mountPath: /data
+          volumes:
+          - name: mypvc
+            persistentVolumeClaim:
+              claimName: pvc-tempest
+        EOF
+      register: tempest_data_pod_create
+      ignore_errors: true
+      tags:
+        - skip_ansible_lint
+    - name: Wait until data pod api pod is deployed
+      command: "kubectl get pod -n openstack tempest-data -o jsonpath='{.status.containerStatuses[].ready}'"
+      register: tempest_data_pod_status
+      until: tempest_data_pod_status.stdout == "true"
+      retries: 60
+      delay: 2
+      changed_when: False
+      when: tempest_data_pod_create.rc == 0
+    - name: Cat the tempest output into a local file
+      shell: "kubectl -n openstack exec -i tempest-data -- cat /data/.stestr/0 > {{ ansible_env.HOME }}/tempest.subunit"
+      register: _tempest_output
+      when: tempest_data_pod_status.rc == 0
+    - name: Fetch the tempest output
+      fetch:
+        dest: "{{ playbook_dir }}/../tempest.subunit"
+        src: "{{ ansible_env.HOME }}/tempest.subunit"
+        flat: yes
+      when:
+        - _tempest_output.rc == 0
+    - name: Delete the data pod
+      command: kubectl -n openstack delete pod tempest-data --grace-period=5
```

Comment:
I would make it force and grace-period of 0. Why waiting, we're gonna crash the whole thing anyway?

Reply:
We don't even have to delete, as it will automatically get destroyed when tearing things down.

Reply:
I dunno to be honest, Im not sure how killing it could affect the pvc attached, so I didnt want to go all hardcore here. If there is a guarantee of not affecting the pvc then by all means lets kill it fast!

Reply:
Yeah, it was only in the case of the refactor above. In that case, after it's fetched, there is nothing else to do.